ee/delivery-hub/pkg/models/event.go

```diff
@@ -231,3 +253,58 @@ func FindLastEventBySourceID(sourceID uuid.UUID) (map[string]any, error) {
 
 	return m, nil
 }
+
+// CompileBooleanExpression compiles a boolean expression.
+//
+// variables: the variables to be used in the expression.
+// expression: the expression to be compiled.
+// caseSensitive: whether the expression should be case sensitive.
+func CompileBooleanExpression(variables map[string]any, expression string, caseSensitive bool) (*vm.Program, error) {
+	options := []expr.Option{
+		expr.Env(variables),
+		expr.AsBool(),
+		expr.WithContext("ctx"),
+		expr.Timezone(time.UTC.String()),
+	}
+
+	if !caseSensitive {
```

Comment:
`caseSensitive` makes it seem like it also applies to the data filter, which is not true. It would be probably better to pass the filter type here directly.